packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -0,0 +1,401 @@
+
+#### _Additional features out of scope of this spec_
+
+-   Allowing the user to tag or mention by entering `@` in the editor and selecting the user name from the drop-down list.
+-   Support for adding images to the editor either by uploading or by pasting it.
+-   Support for adding hyperlinks to the existing words in the editor. However, adding raw links by pasting them to the editor is supported for the
+    initial release.
+-   Support for [striking out](https://tiptap.dev/api/marks/strike) and [underlining](https://tiptap.dev/api/marks/underline) text. We use the
+    [prosemirror-markdown](https://github.com/ProseMirror/prosemirror-markdown) serializer and parser to convert the text into markdown format and vice
+    versa. However, the supported functionality of prosemirror-markdown, as mentioned in their
+    [documentation](https://github.com/ProseMirror/prosemirror-markdown#documentation), does not include support for a strikeout and underline. To
+    address this limitation, we need to extend the class and add the necessary nodes to both the serializer and parser which will be added in the
+    subsequent releases.
+-   Provide APIs to configure visibility/disabled state of toolbar buttons for a rich text editor.
+
+### Risks and Challenges
+
+-   Due to immediate requirements for comments feature from a business customer, any additional enhancements or requirements apart from whatever is
+    mentioned in this spec are deferred to future scope.
+-   Currently, we will begin by referring to the existing
+    [Interaction design workflow](https://www.figma.com/proto/Q5SU1OwrnD08keon3zObRX/SystemLink?type=design&node-id=6280-94118&scaling=min-zoom&page-id=2428%3A32954&starting-point-node-id=6280%3A94118&show-proto-sidebar=1)
+    of the comments feature. Once the visual design for these components is complete, we will then be implementing those specific changes within the defined
+    scope of development. However, we will still make use of existing nimble components such as `nimble-button` and `nimble-text-area` to maintain a
+    consistent design for the initial release.
+
+### Prior Art/Examples
+
+Sample Mockup Screenshots
+
+_`nimble-rich-text-editor`_
+
+![Editor](./spec-images/editor-sample.png)
+
+_`nimble-rich-text-viewer`_
+
+![Viewer](./spec-images/viewer-sample.png)
+
+## Design
+
+### `nimble-rich-text-editor`
+
+The `nimble-rich-text-editor` will be divided into two sections namely an `editor` section and a `footer` section.
+
+1. `editor` section is the actual text area to add or update rich text content.
+2. `footer` section consists of `nimble-button` to control each text formatting functionalities like bold, italic, etc, and a
+   `footer-actions` slot element which is typically used to add action buttons to the right bottom of the component.
+
+Example usage of the `nimble-rich-text-editor` in the application layer is as follows:
+
+```html
+<nimble-rich-text-editor>
+    <div slot="footer-actions">
```

Comment:
Is it expected that a client would provide a `div` containing multiple buttons like this, or would they do something more like:
```
<nimble-rich-text-editor>
    <nimble-button slot="footer-actions>Cancel</nimble-button>
    <nimble-button slot="footer-actions>Add Comment><nimble-button>
</nimble-rich-text-editor>
```

In the example you provide, it would be the responsibility of a client to space the buttons appropriately. However, in the example I provided, the nimble component could add the appropriate spacing between each component placed in the `footer-actions` slot.

Reply:
I have updated the spec as per the example you have showed. The standard spacing between the slotted elements will be handled by the component in `footer-actions` sections and other styling will be handled by the client.